weatherAplication/Client/ClientAPI.swift

```diff
@@ -0,0 +1,100 @@
+import SnapKit
+import UIKit
+
+final class ClientAPI {
+    private var kElementsToShow = 24
+    private let kCharsToDrop = 11
+    private let urlGeoDBString: String = "http://geodb-free-service.wirefreethought.com/v1/geo/cities?"
+    private let urlOpenMeteoString: String = "https://api.open-meteo.com/v1/forecast?"
+    static let sharedInstance: ClientAPI = {
+        let instance = ClientAPI()
+        return instance
+    }()
+
+    private init() {}
+
+    func searchWeather(latitude: Double, longitude: Double, view: UIViewController, completion: @escaping (_ result: [WeatherData], _ date: String) -> Void) {
+        guard let queryURL = URL(string: urlOpenMeteoString + "latitude=\(latitude)&longitude=\(longitude)&hourly=temperature_2m,surface_pressure,weathercode,windspeed_10m") else { return }
+        let session = URLSession.shared
+
+        session.dataTask(with: queryURL, completionHandler: { [weak self] data, response, error -> Void in
+
+            if error != nil {
+                let alert = UIAlertController(title: "Error", message: error?.localizedDescription, preferredStyle: .alert)
+                alert.addAction(UIAlertAction(title: "OK", style: .default))
+                view.present(alert, animated: true)
+            }
```

Comment:
To powinno być zrobione w taki sposób, że funkcja jest oznaczona że może zwrócić error i zrobił handlowanie erroru dopiero w viewControllerze